src/ucs/sys/sock.c

```diff
@@ -311,8 +315,11 @@ ucs_socket_handle_io_error(int fd, const char *name, ssize_t io_retval, int io_e
     ucs_status_t status;
 
     if (io_retval == 0) {
+        /* 0 can be returned only by recv() system call as an error if
```

Comment:
i think 0 can also be returned if len==0 for send(). so we can add 'len' parameter to handle_io_error.

Reply:
i think it should be in this function
we must concentrate err check logic in one place and not split it to multiple places

Reply:
> we must concentrate err check logic in one place and not split it to multiple places

yes, but the check for `0` when data length is `0` - not an error handling, it's expected behavior when a user is trying to transfer 0-byte messages (e.g. notification mechanism to wakeup peer that waits on a `select()`/`poll()`/`epoll_wait()`)